File: LeetCode/ProblemSet/Volume (3700-3799)/LeetCode3754/LeetCode3754-Solution-01.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        if n == 0:
            return 0
        
        x, sum, k = 0, 0, 1
        while n > 0:
            n, v = divmod(n, 10)

            if v == 0:
                continue

            sum += v
            x += v * k
            k *= 10

        return x * sum 

File: LeetCode/ProblemSet/Volume (3700-3799)/LeetCode3754/LeetCode3754-Solution-01.cpp
#include <bits/stdc++.h>

class Solution {
public:
    long long sumAndMultiply(int n) {
        if (n == 0) {
            return 0;
        }

        long long x = 0, sum = 0, k = 1;
        while (n > 0) {
            int v = n % 10;
            n /= 10;
            if (v == 0) {
                continue;
            }

            sum += v;
            x += v * k;
            k *= 10;
        }

        return x * sum;
    }
};